docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: docs
+title: Best Practices
+permalink: /documentation/recipes/best-practices/best-practices.html
+---
+# Best Practices
+
+This article covers the recommended ways to test with TestCafe and covers the following topics:
+
+* Smart Assertions
+* Use of Page Objects
+* Use of Roles for Login
+* File structure
+* Clean up between Tests (beforeEach)
+* E2E Tests Limitations
+* Selectors strategy
+
+## Smart Assertions
+
+In end-to-end web testing, peripheral factors (like network lag, processor, or memory bottlenecks in containers) can interfere with the assertions and render tests inconclusive (otherwise known as 'flaky').
+
+Out of the box, TestCafe includes a [Smart Assertion Query Mechanism](../../guides/basic-guides/assert.md#smart-assertion-query-mechanism), which helps mitigate these effects. This mechanism introduces additional wait time for all the assertions: if an assertion fails, the test doesn't fail straight away. Instead, the assertion retries multiple times within a timeout. That helps to stabilize the tests without a big impact on performance.
+
+However, the smart assertions query mechanism only works with promises returned from TestCafe's own `Selector API`.
+
+The following example demonstrates the potential issue:
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with Selector', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    //this technique leads to unstable results. Avoid it.
+    const developerName      = await Selector('#developer-name').value;
+    //
+
+    console.log(developerName);
+
+    await t
+        .wait(1000)
+        .expect(developerNameInput.value).eql('')
+        .typeText(developerNameInput, 'Peter')
+        .expect(developerName).eql('Peter');
+});
+```
+
+In this snippet, the `developerName` is initialized with the value of a Selector, but because of the `await` keyword, the value is calculated once and doesn't ever update. This effectively disables the smart assertions query mechanism, which in this case leads to inconclusive results. This test fails.
```

Comment:
> ever

we can omit it for good, just *doesn't update* - fine for tech docs

> This effectively disables

This kind of scientific wording is good when you are researching or exploring something, like a natural phenomenon. It's can be appropriate in IT when you review someone else's product (this is also research and exploration, kinda). But when you are documenting your own product, saying things like "this effectively disables" seems out of place. You should be more specific and certain.

In this case, I suspect the mechanism doesn't activate at all if you are not passing a Promise, doesn't it? So, you can simply say *it disables*.